PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/core/FunctionUtils.java

```diff
@@ -0,0 +1,375 @@
+import com.microsoft.azure.common.function.bindings.BindingEnum;
+import com.microsoft.azure.common.function.configurations.FunctionConfiguration;
+import com.microsoft.azure.common.function.utils.CommandUtils;
+import com.microsoft.azure.functions.annotation.StorageAccount;
+import com.microsoft.azure.maven.common.utils.SneakyThrowUtils;
+import org.apache.commons.collections4.ListUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.idea.maven.project.MavenProject;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public class FunctionUtils {
+
+    private static final String FUNCTION_JSON = "function.json";
+    public static final String FUNCTION_JAVA_LIBRARY_ARTIFACT_ID = "azure-functions-java-library";
+    private static final String AZURE_FUNCTION_ANNOTATION_CLASS = "com.microsoft.azure.functions.annotation.FunctionName";
+    private static final String HTTP_OUTPUT_DEFAULT_NAME = "$return";
+    private static final String DEFAULT_HOST_JSON = "{\"version\":\"2.0\",\"extensionBundle\":" +
+            "{\"id\":\"Microsoft.Azure.Functions.ExtensionBundle\",\"version\":\"[1.*, 2.0.0)\"}}\n";
+    private static final String DEFAULT_LOCAL_SETTINGS_JSON = "{ \"IsEncrypted\": false, \"Values\": " +
+            "{ \"FUNCTIONS_WORKER_RUNTIME\": \"java\" } }";
+
+    public static Module[] listFunctionModules(Project project) {
+        final Module[] modules = ModuleManager.getInstance(project).getModules();
+        return Arrays.stream(modules).filter(m -> {
+            final GlobalSearchScope scope = GlobalSearchScope.moduleWithLibrariesScope(m);
+            final PsiClass ecClass = JavaPsiFacade.getInstance(project).findClass(AZURE_FUNCTION_ANNOTATION_CLASS, scope);
+            return ecClass != null;
+        }).toArray(Module[]::new);
+    }
+
+    public static Module getFunctionModuleByFilePath(Project project, String moduleFilePath) {
+        final Module[] modules = listFunctionModules(project);
+        return Arrays.stream(modules)
+                .filter(module -> StringUtils.equals(moduleFilePath, module.getModuleFilePath()))
+                .findFirst().orElse(null);
+    }
+
+    public static boolean isFunctionProject(Project project) {
+        final List<Library> libraries = new ArrayList<>();
+        OrderEnumerator.orderEntries(project).productionOnly().forEachLibrary(library -> libraries.add(library));
+        return ListUtils.indexOf(libraries, library ->
```

Comment:
move StringUtils.contains(library.getName(), FUNCTION_JAVA_LIBRARY_ARTIFACT_ID
to forEachLibrary

Reply:
Fixed